tests/conftest.py

```diff
@@ -116,3 +118,49 @@ def init_sqldb(dsn):
     engine = create_engine(dsn)
     Base.metadata.create_all(bind=engine)
     return sessionmaker(bind=engine)
+
+
+class DictToObjectWrapper:
+    """
+    This class enables to wrap dict in a object to enable attribute access. e.g.
+    >>> d = {'a': 1, 'b': {'c': 2}, 'd': ["hi", {'foo': "bar"}]}
+    >>> x.b.c
+    2
+    >>> x.d[1].foo
+    'bar'
+
+    Notes:
+        - changing the values of the dict after initialization won't affect the class instance attribute values
+        - changing the class instance attribute values after initialization won't affect dict
+    """
+
+    def __init__(self, dictionary):
+        for key, value in dictionary.items():
+            if isinstance(value, (list, tuple)):
+                setattr(
+                    self,
+                    key,
+                    [
+                        DictToObjectWrapper(item) if isinstance(item, dict) else item
+                        for item in value
+                    ],
+                )
+            else:
+                setattr(
+                    self,
+                    key,
+                    DictToObjectWrapper(value) if isinstance(value, dict) else value,
+                )
+
+
+class DictToK8sObjectWrapper(DictToObjectWrapper):
```

Comment:
?

Reply:
This is extending the generalized `DictToObjectWrapper` with functionality only mocked k8s objects should have (like `to_dict()` method)

Reply:
Neither this nor `DictToObjectWrapper` mention "mock" or inherit from Mock object.
If they are meant to be used for mocking they should inherit from Mock and named as such I suppose 🤷

Reply:
added

Reply:
Talked offline, this whole thing isn't needed, I should have simply used the k8s client classes instead of tranforming dict to mock objects